rfcs/01924_tls_attestation_ca.md

```diff
@@ -0,0 +1,161 @@
+same time trust in this application without directly attesting it.
+
+### Attestation Workflow
+
+The Proxy Attestation Service has 4 main interacting components:
+
+- **Certificate Authority (CA)**
+  - Separate application that implements Proxy Attestation Service
+  - Remotely attest backend applications and can be remotely attested by clients
+  - Creates signed certificates for backend applications
+  - Distributes its root certificate among the clients
+  - Runs inside the TEE environment
+- **Server**
+  - Backend application that processes requests from TLS clients
+  - Runs inside the TEE environment
+- **Client**
+  - Client application that connects to applications using TLS
+  - Runs on a user device or inside of a browser
+- **Provider**
+  - TEE platform provider that can confirm authenticity of TEE platform keys
+  - It’s important to note that the Provider is an external server (i.e. belongs
+    to AMD)
+
+It is assumed that prior to the attestation **Server** remotely attests its TEE
+platform key with the **Provider**. **Provider** sends back a signature rooted
+in some well-known root key (e.g. AMD root key), thus confirming the
+authenticity of TEE platform keys.
+
+#### Workflow
+
+The detailed workflow of the remote attestation is represented by 2 stages.
+
+First stage corresponds to the backend application attestation, which happens
+only once when the backend application starts:
+
+1. **Server** connects to the **CA**
+1. **CA** attests the **Server** using a remote attestation protocol
+   corresponding to the Server’s TEE
+   1. When requesting a TEE report from the TEE firmware, **Server** also puts
+      the hash of its TLS public key into it
+1. **Server** sends **CA** a certificate signing request
+   1. Communication happens over a secure channel creates as part of the remote
+      attestation
+1. **CA** creates a signed certificate and sends it back to the **Server**
+   1. The certificate contains **Server**'s TEE report
+
+<!-- From: -->
+<!-- https://sequencediagram.googleplex.com/view/6235412701904896 -->
+<img src="images/tls_attestation_ca_workflow_server.png" width="850">
+
+Second stage corresponds to the client connection:
+
+1. **Client** remotely attests the **CA**
+1. **CA** sends **Client** its root certificate
+   1. **Client** trusts that this is a correct certificate, since it was sent
+      via a secure channel created during the attestation process
+   1. **Client** needs to request the root certificate only once at startup, and
+      then the client can use it for all **Server** connections
+1. **Client** connects to the **Server** using TLS
+1. **Server** sends **Client** a TLS certificate previously signed by **CA**
+1. **Client** verifies that the TLS certificate provided by the **Server** was
+   signed by the **CA**’s root certificate
+   1. **Client** also verifies the TEE report provided in the **Server**’s
+      certificate
```

Comment:
This is the important step, right? Should it mention how to actually verify the report? in particular, what about the public key hash that was in the TEE report?